pkg/object/mqttproxy/broker.go

```diff
@@ -173,19 +182,24 @@ func (b *Broker) setListener() error {
 	return err
 }
 
-func (b *Broker) reconnectWatcher() {
+func (b *Broker) reconnectOnlyDeleteWatcher() {
 	if b.closed() {
 		return
 	}
 
-	ch, cancelFunc, err := b.sessMgr.store.watchDelete(sessionStoreKey(""))
-	if err != nil {
+	var ch <-chan map[string]*string
+	var cancelFunc func()
+	var err error
+
+	for {
+		ch, cancelFunc, err = b.sessMgr.store.watchDelete(sessionStoreKey(""))
+		if err == nil {
+			break
+		}
 		logger.SpanErrorf(nil, "get watcher for session failed, %v", err)
 		time.Sleep(10 * time.Second)
-		go b.reconnectWatcher()
-		return
 	}
-	go b.watchDelete(ch, cancelFunc)
+	go b.onlyWatchDelete(ch, cancelFunc)
```

Comment:
the code below this line could also be called concurrently, please make sure this is not an issue.